File: backend/app/schemas/ai.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel


class AIExplainRequest(BaseModel):
    client_nom: str
    client_code: Optional[str] = None
    risk_score: int
    facteurs: List[str]
    alerte_ref: Optional[str] = None
    anomaly_score: Optional[float] = None
    predicted_risk: Optional[str] = None


class AIExplainResponse(BaseModel):
    synthese: str
    points_cles: List[str]
    rappel_conformite: str = "⚠️ La décision finale de blocage ou de déclaration CENTIF revient exclusivement à l'analyste habilité (IA-03)."
    source_moteur: str = "LAKANA-Rules-Engine"


class AIChatRequest(BaseModel):
    message: str


class AIChatResponse(BaseModel):
    response: str
    intent: str
    suggestions: List[str]
    context_client: Optional[Dict[str, Any]] = None
    points_cles: Optional[List[str]] = None
    source_moteur: str = "Assistant IA LAKANA (Hybride Règles Métier + ML)"
    rappel_conformite: str = "⚠️ Rappel : la décision finale revient à l'analyste habilité (IA-03)."


class AIContextResponse(BaseModel):
    top_client: Optional[Dict[str, Any]] = None
    stats_alertes: Dict[str, Any]
    stats_clients: Dict[str, Any]
    models_ready: bool
    suggested_queries: List[str]

File: LAKANA-/frontend/src/models/ai.ts
export interface AIExplainRequest {
  client_nom: string
  client_code?: string
  risk_score: number
  facteurs: string[]
  alerte_ref?: string
}

export interface AIExplainResponse {
  synthese: string
  points_cles: string[]
  rappel_conformite: string
  source_moteur: string
}

export interface AIChatRequest {
  message: string
}

export interface AIChatResponse {
  response: string
  intent: string
  suggestions: string[]
  context_client?: {
    id: string
    nom: string
    prenom?: string
    code_client: string
    risk_score: number
    niveau_risque: string
    facteurs: string[]
    decomposition?: Record<string, { points: number; max: number }>
    transactions_count?: number
    est_ppe?: boolean
    profession?: string
  }
  points_cles?: string[]
  source_moteur: string
  rappel_conformite: string
}

export interface AIContextResponse {
  top_client?: {
    id: string
    nom: string
    prenom?: string
    code_client: string
    risk_score: number
    niveau_risque: string
    facteurs: string[]
    decomposition?: Record<string, { points: number; max: number }>
    transactions_count?: number
    est_ppe?: boolean
  }
  stats_alertes: {
    total: number
    bloquantes: number
    analyser: number
    informatives: number
    cloturees: number
    taux_resolution: number
  }
  stats_clients: {
    total: number
    eleve: number
    moyen: number
    faible: number
  }
  models_ready: boolean
  suggested_queries: string[]
}

export interface MLPredictResponse {
  client_id: string
  client_nom: string
  code_client?: string
  anomaly_score: number
  is_anomaly: boolean
  predicted_risk: string
  confidence: number
  model_used: string
  fallback: boolean
  facteurs_ia: string[]
  features: Record<string, number>
}

export interface MLTrainResponse {
  status: string
  total_samples: number
  real_clients_count: number
  synthetic_samples: number
  features_count: number
  features: string[]
  models_saved: string[]
}

